src/System.IO.Compression/src/System/IO/Compression/DeflateStream.Unix.cs

```diff
@@ -11,17 +11,18 @@ private static WorkerType GetDeflaterType()
         {
             try
             {
-                // Make any P/Invoke into zlib to ensure we're able to find and use it.
-                // If we are, then use zlib.
-                Interop.zlib.zlibCompileFlags(); 
-                return WorkerType.ZLib;
+                if (Interop.zlib.IsZLibAvailable())
+                {
+                    return WorkerType.ZLib;
+                }
             }
             catch
```

Comment:
If we're going to call into our own shim, we no longer need the try/catch, which was there to catch the P/Invoke exception that would arise from the export not being available.